File: CS50_assignments/pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long cardnumber = get_long("Number: ");
    int count = 0;
    int products_digits_sum = 0;
    int digits_sum = 0;
    int digits_not_multiplied_sum = 0;
    int digit;
    int checksum = 0;
    int seconddigit = 0;

    while (cardnumber > 0)
    {
        digit = cardnumber % 10;
        count++;
        cardnumber = cardnumber / 10;


        if (cardnumber < 10 && cardnumber > 0)
        {

            seconddigit = digit;

        }

        if (count % 2 == 0)
        {
            int copydigit = digit;
            copydigit = copydigit * 2;

            products_digits_sum = 0;
            while (copydigit > 0)
            {
                products_digits_sum += copydigit % 10;

                copydigit = copydigit / 10;
            }

            digits_sum += products_digits_sum;

        }
        else {
            digits_not_multiplied_sum += digit;
        }

    }


    checksum = digits_sum + digits_not_multiplied_sum;



    if ((checksum % 10) == 0)
    {


        if (count == 15 && digit == 3 && (seconddigit == 4 || seconddigit == 7))
            printf("AMEX\n");
        else if (count == 16 && digit == 5 && seconddigit < 6)
            printf("MASTERCARD\n");
        else if ((count == 13 || count == 16) && digit == 4)
            printf("VISA\n");
        else
            printf("INVALID\n");
    }
    else
        printf("INVALID\n");

}

File: CS50_assignments/week6/sentimental-credit/credit.py
from cs50 import get_int


def main():
    while True:
        credit_card = get_int("Number..")
        if credit_card >= 0:
            break

    if check_validity(credit_card):
        print_card_brand(credit_card)
    else:
        print("INVALID")


def check_validity(ccn):
    return checksum(ccn)

def checksum(ccn):
    sum = 0

    for i in range(len(str(ccn))):
        if (i % 2 == 0):
            sum += ccn % 10
        else:
            digit = 2 * (ccn % 10)
            sum += digit//10 + digit%10

        ccn //= 10

    return sum % 10 == 0

def print_card_brand(ccn):
    if (ccn >= 34e13 and ccn < 35e13) or (ccn >= 37e13 and ccn < 38e13):
        print("AMEX")
    elif (ccn >= 51e14 and ccn < 56e14):
        print("MASTERCARD")
    elif ((ccn >= 4e12 and ccn < 5e12) or (ccn >= 4e15 and ccn < 5e15)):
        print("VISA")
    else:
        print("INVALID")

main()



